### ISSUE TITLE
bst 1.x overnigth tests are failing: Message handling out of sync

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#1098]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/1098)
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Aug 5, 2019, 18:03

Job [#266132777](https://gitlab.com/BuildStream/buildstream/-/jobs/266132777) failed for 63991dc8c7383a4ba13ea6ad534c4079581e9a37:

cc [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb)

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 6, 2019, 23:13

I was unable to reproduce this locally unfortunately, I'm building `bootstrap/gcc.bst` and didn't run into any issue with `bootstrap/build/image.bst`.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tpollard]](https://gitlab.com/tpollard) on Aug 7, 2019, 17:38

The message handling bug shouldn't be causing the actual build to fail, from what I can tell it attempted to pull an artifact for `bootstrap/build/base-sdk/image.bst` which it couldn't find in the cache, so defaulted to fetching the source from flathub which later failed with the last report being:

`[fetch:bootstrap/build/base-sdk/image.bst] STATUS  Receiving objects: 7% (834/11676) 1.3 MB`

Maybe the fetch timed out, in !1500 I should have hopefully mitigated the sync bug you're seeing though, so we can actually get the output of things such as this failing

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on Aug 30, 2019, 12:46

[[Gitlab user @tpollard]](https://gitlab.com/tpollard) any idea what is the timeout of a normal fetch?, the same CI for 1.2 (with no cache, so it has to download everything) seems to work fine: https://gitlab.com/BuildStream/buildstream/-/jobs/283637436

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 30, 2019, 17:27

mentioned in commit 235ed63a69d247218448b76f603a17652c7f0d09

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 30, 2019, 17:40

This is fixed by 235ed63a, which simply reverts 08ead05bbe24a6de94e26b1121e5aa8dd341f691

It is hard to say exactly what is wrong with the installation, but when running `pip3 install -r requirements/requirements.txt -r requirements/plugin-requirements.txt`, we end up getting a stack trace while installing PyGObject, this appears to have been happening for a while without causing CI to fail.

The change to a fedora 30 image caused the `ostree` element to consistently raise this error, while on the regular debian 9, the `ostree` element still works fine, the stack trace at install time is however present in both.

For the record, here is the stack trace installing PyGObject:

```
Collecting Click==7.0 (from -r requirements/requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl (81kB)
Collecting grpcio==1.18.0 (from -r requirements/requirements.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/ea/72/dbdce5ba0eb2ee22069430f1c18c2fef22b4f6ffbf188ab2d416695f3992/grpcio-1.18.0-cp35-cp35m-manylinux1_x86_64.whl (10.6MB)
Collecting pluginbase==0.7 (from -r requirements/requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/6e/f4/1db0a26c1c7fad81a1214ad1b02839a7bd98d8ba68f782f6edcc3d343441/pluginbase-0.7.tar.gz (40kB)
Collecting protobuf==3.6.1 (from -r requirements/requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/bf/d4/db7296a1407cad69f043537ba1e05afab3646451a066ead7a314d8714388/protobuf-3.6.1-cp35-cp35m-manylinux1_x86_64.whl (1.1MB)
Collecting psutil==5.4.8 (from -r requirements/requirements.txt (line 5))
  Downloading https://files.pythonhosted.org/packages/e3/58/0eae6e4466e5abf779d7e2b71fac7fba5f59e00ea36ddb3ed690419ccb0f/psutil-5.4.8.tar.gz (422kB)
Collecting ruamel.yaml==0.15.51 (from -r requirements/requirements.txt (line 6))
  Downloading https://files.pythonhosted.org/packages/45/34/4818c6936fbe2a8dbda9430eb50d7e2e035e24db0ee5b232c68c54cb0c44/ruamel.yaml-0.15.51-cp35-cp35m-manylinux1_x86_64.whl (620kB)
Collecting setuptools==40.6.2 (from -r requirements/requirements.txt (line 7))
  Downloading https://files.pythonhosted.org/packages/e7/16/da8cb8046149d50940c6110310983abb359bbb8cbc3539e6bef95c29428a/setuptools-40.6.2-py2.py3-none-any.whl (573kB)
Collecting Jinja2==2.10 (from -r requirements/requirements.txt (line 9))
  Downloading https://files.pythonhosted.org/packages/7f/ff/ae64bacdfc95f27a016a7bed8e8686763ba4d277a78ca76f32659220a731/Jinja2-2.10-py2.py3-none-any.whl (126kB)
Collecting MarkupSafe==1.1.0 (from -r requirements/requirements.txt (line 10))
  Downloading https://files.pythonhosted.org/packages/3e/a5/e188980ef1d0a4e0788b5143ea933f9afd760df38fec4c0b72b5ae3060c8/MarkupSafe-1.1.0-cp35-cp35m-manylinux1_x86_64.whl
Requirement already satisfied: six==1.12.0 in /usr/local/lib/python3.5/dist-packages (from -r requirements/requirements.txt (line 11))
Collecting arpy==1.1.1 (from -r requirements/plugin-requirements.txt (line 1))
  Downloading https://files.pythonhosted.org/packages/d2/15/09fbfc0d43b66a3845098377249539aa44edcd541e17eabe2e11b4e3855e/arpy-1.1.1.tar.gz
Collecting PyGObject==3.30.4 (from -r requirements/plugin-requirements.txt (line 2))
  Downloading https://files.pythonhosted.org/packages/8c/1f/76533985b054473ef6ab1ba4d9c00d62da502f8b43d3171ae588ec81ae93/PyGObject-3.30.4.tar.gz (690kB)
Collecting pycairo==1.18.0 (from -r requirements/plugin-requirements.txt (line 4))
  Downloading https://files.pythonhosted.org/packages/a6/54/23d6cf3e8d8f1eb30e0e58f171b6f62b2ea75c024935492373639a1a08e4/pycairo-1.18.0.tar.gz (202kB)
Building wheels for collected packages: pluginbase, psutil, arpy, PyGObject, pycairo
  Running setup.py bdist_wheel for pluginbase: started
  Running setup.py bdist_wheel for pluginbase: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/ce/78/59/1c57c1f57381a115b31c12c969baa20f62b97f7a64c079cce3
  Running setup.py bdist_wheel for psutil: started
  Running setup.py bdist_wheel for psutil: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/d2/71/40/9c6993129f8cda369d0f21c46a13a6adab7fb1664fe6512551
  Running setup.py bdist_wheel for arpy: started
  Running setup.py bdist_wheel for arpy: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/5e/39/a4/230621c4dce61510a12f05a6114e58b0f4c437b5def437d4a0
  Running setup.py bdist_wheel for PyGObject: started
  Running setup.py bdist_wheel for PyGObject: finished with status 'error'
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-owpvep97/PyGObject/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpa6zjvikkpip-wheel- --python-tag cp35:
  Failed building wheel for PyGObject
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.5
  creating build/lib.linux-x86_64-3.5/pygtkcompat
  copying pygtkcompat/generictreemodel.py -> build/lib.linux-x86_64-3.5/pygtkcompat
  copying pygtkcompat/pygtkcompat.py -> build/lib.linux-x86_64-3.5/pygtkcompat
  copying pygtkcompat/__init__.py -> build/lib.linux-x86_64-3.5/pygtkcompat
  creating build/lib.linux-x86_64-3.5/gi
  copying gi/_error.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/docstring.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_ossighelper.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/importer.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_propertyhelper.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_gtktemplate.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_option.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_signalhelper.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_constants.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/module.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/pygtkcompat.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/types.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/__init__.py -> build/lib.linux-x86_64-3.5/gi
  copying gi/_compat.py -> build/lib.linux-x86_64-3.5/gi
  creating build/lib.linux-x86_64-3.5/gi/repository
  copying gi/repository/__init__.py -> build/lib.linux-x86_64-3.5/gi/repository
  creating build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/GLib.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/Gdk.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/GdkPixbuf.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/Gtk.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/Gio.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/Pango.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/keysyms.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/GObject.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/__init__.py -> build/lib.linux-x86_64-3.5/gi/overrides
  copying gi/overrides/GIMarshallingTests.py -> build/lib.linux-x86_64-3.5/gi/overrides
  running build_ext
  pycairo: new API
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 1241, in <module>
      main()
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 1236, in main
      zip_safe=False,
    File "/usr/local/lib/python3.5/dist-packages/setuptools/__init__.py", line 145, in setup
      return distutils.core.setup(**attrs)
    File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/usr/local/lib/python3.5/dist-packages/wheel/bdist_wheel.py", line 188, in run
      self.run_command('build')
    File "/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/usr/lib/python3.5/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/lib/python3.5/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 1076, in run
      self._setup_extensions()
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 1071, in _setup_extensions
      add_pycairo(gi_cairo_ext)
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 1055, in add_pycairo
      ext.include_dirs += [get_pycairo_include_dir()]
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 877, in get_pycairo_include_dir
      include_dir = find_path(find_new_api())
    File "/tmp/pip-build-owpvep97/PyGObject/setup.py", line 822, in find_new_api
      import cairo
  ImportError: No module named 'cairo'
  
  ----------------------------------------
  Running setup.py clean for PyGObject
  Running setup.py bdist_wheel for pycairo: started
  Running setup.py bdist_wheel for pycairo: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/e6/bb/aa/7b71758ea95e5ae6f1aef55bbc34cc4dd236383dc6554af0c5
Successfully built pluginbase psutil arpy pycairo
Failed to build PyGObject
Installing collected packages: Click, grpcio, pluginbase, setuptools, protobuf, psutil, ruamel.yaml, MarkupSafe, Jinja2, arpy, pycairo, PyGObject
  Found existing installation: setuptools 40.8.0
    Uninstalling setuptools-40.8.0:
      Successfully uninstalled setuptools-40.8.0
  Found existing installation: MarkupSafe 0.23
    Not uninstalling markupsafe at /usr/lib/python3/dist-packages, outside environment /usr
  Running setup.py install for PyGObject: started
    Running setup.py install for PyGObject: finished with status 'done'
Successfully installed Click-7.0 Jinja2-2.10 MarkupSafe-1.1.0 PyGObject-3.30.4 arpy-1.1.1 grpcio-1.18.0 pluginbase-0.7 protobuf-3.6.1 psutil-5.4.8 pycairo-1.18.0 ruamel.yaml-0.15.51 setuptools-40.6.2
```

Note that this revert does not cause the `git_tag` bug #1097 to come back, this seems to also be fixed by the update to a more recent `bst-external` in the tests.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 30, 2019, 17:40

closed
